### PR TITLE
DOCS-15709 collection level filter

### PR DESCRIPTION
**Pull request by davemungo**
[PAGE](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCS-15709-collection-level-filter-v1.0/reference/collection-level-filtering/)
[LIMITATIONS](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCS-15709-collection-level-filter-v1.0/reference/limitations/#collection-level-filtering)
[START API](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCS-15709-collection-level-filter-v1.0/reference/api/start/)

[JIRA](https://jira.mongodb.org/browse/DOCS-15709)

### CODE REVIEW

File: source/reference/collection-level-filtering.txt [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+specify the entire ``books`` database in your filter. The filter will
+not work if you only specify the ``authors`` collection.
+
+This filter works:
+
+.. code-block:: javascript
+
+   "includeNamespaces": [
+     {
+       “database": "books"
```
Comment (kennethdyer) [inline source/reference/collection-level-filtering.txt:175]: Pygments is producing weirdness in the syntax highlighting for this and the next code block.  I'd suggest switching from `javascript` to `json`.

Also, it looks like you caught a smart quote at the start the `database` field.

File: source/reference/collection-level-filtering.txt [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+     }
+   ]
+
+This filter does not work with ``$out``:
+
+.. code-block:: javascript
+
+   "includeNamespaces": [     
+     {
+       “database": "books",
```
Comment (kennethdyer) [inline source/reference/collection-level-filtering.txt:185]: Smart quote.

**Review by kennethdyer — CHANGES_REQUESTED** [review]
LGTM.  I left a few comments.

**Review by davemungo — COMMENTED** [review]
@kennethdyer Thanks for your suggestions, send on to external review next

File: source/reference/collection-level-filtering.txt [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      :ref:`/start <c2c-api-start>` API call to begin syncing.
+
+      .. literalinclude:: /includes/api/requests/start-filtered.sh
+         :language: shell
+
+For an example configuration, see: :ref:`c2c-example-start-with-filter`.
+
+.. _c2c-change-filter:
+
+Change the Filter
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:76]: Can we add limitations to this header somewhere?

File: source/reference/collection-level-filtering.txt [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+.. _c2c-configure-filter:
+
+Configure a Filter
+------------------
+
+.. important::
+
+   Once you start ``mongosync`` with a filter in place, the filter
+   cannot be modified directly. If you do need to modify the filter,
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:32]: I'd consider changing `If you do need to modify the filter` to `If you misconfigured a filter`. I don't want to give people the idea that filtered sync should currently be expected to be used in a mid migration filter changing manner

File: source/includes/api/facts/includeNamespaces-syntax.rst [hunk 5]
```diff
@@ -0,0 +1,14 @@
+.. code-block:: javascript
+
+   [ {
```
Comment (rohshar) [inline source/includes/api/facts/includeNamespaces-syntax.rst:3]: Can you put the first curly brace on its own line

File: source/reference/collection-level-filtering.txt [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+      The filter syntax is: 
+
+      .. include:: /includes/api/facts/includeNamespaces-syntax.rst
+
+      Create an entry in the ``includeNamespaces`` array for each
+      database that you identified in step 1. Use the ``"database"``
+      field to specify the database name.
+      
+      If you want to filter on collections within a database, add those
+      collections to the ``"collections"`` field for that database
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:61]: to a list in the collections field?

File: source/reference/collection-level-filtering.txt [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+         :language: shell
+
+For an example configuration, see: :ref:`c2c-example-start-with-filter`.
+
+.. _c2c-change-filter:
+
+Change the Filter
+-----------------
+
+You cannot update an existing filter. You must remove the old filter and
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:79]: There is no notion of removing the old filter. Can you rephrase this to say something along the lines of stop the old migration, clean up data, and restart?

File: source/reference/collection-level-filtering.txt [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+
+   .. step:: Remove user databases.
+
+      a. List the databases in the cluster
+
+         .. code-block:: javascript
+          
+            show databases
+
+      #. Remove user databases. The ``admin`` and ``config`` databases
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:115]: local is also a system DB users shouldn't drop.

File: source/reference/collection-level-filtering.txt [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+         Repeat this step for each user database list:
+
+         .. code-block:: javascript
+          
+            use <user database name>
+            db.dropDatabase()
+            db.dropDatabase()
+
+         Note: After the first :method:`db.dropDatabase()` operation
+         completes, you need to run it a second time to remove the
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:129]: I'm not familiar with this. I typically only drop the database once, and it works.

File: source/reference/collection-level-filtering.txt [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+Follow these steps to prepare the destination cluster for a new filter.
+
+.. procedure::
+   :style: normal
+
+   .. step:: Remove ``mongosync_reserved_for_internal_use``.
+
+      a. Stop the ``mongosync`` process. 
+      
+      #. Connect to the :binary:`mongos` instance in the destination
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:96]: Filtered sync also works on replica sets, can you add that you can connect to the primary mongod

File: source/reference/collection-level-filtering.txt [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+specify the entire ``books`` database in your filter. The filter will
+not work if you only specify the ``authors`` collection.
+
+This filter works:
+
+.. code-block:: json
+
+   "includeNamespaces": [
+     {
+       "database": "books"
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:175]: I'd call the database literature or something, its confusing if both a collection and DB have the same name.

File: source/reference/collection-level-filtering.txt [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+   ]
+
+This filter does not work with ``$out``:
+
+.. code-block:: json
+
+   "includeNamespaces": [
+     {
+       "database": "books",
+       "collections": [ "authors" ]   // DOES NOT WORK WITH $OUT
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:186]: Include the books collection here, showing that having both books and authors is still not enough.

File: source/reference/collection-level-filtering.txt [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+
+      db.books.aggregate( [
+         { $group : { _id : "$author", books: { $push: "$title" } } },
+         { $out : "authors" }
+      ] )
+
+The ``$out`` stage creates the ``authors`` collection in the ``books``
+database. If you want to include the ``authors`` collection, you must
+specify the entire ``books`` database in your filter. The filter will
+not work if you only specify the ``authors`` collection.
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:167]: only specify the authors and books collections

File: source/includes/limitations-filtering.rst [hunk 14]
```diff
@@ -0,0 +1,20 @@
+- Filtering is not supported with :ref:`reversible sync
+  <c2c-api-reverse>`.
+- The destination cluster must not contain user data prior to starting.
+- The destination cluster must not contain the
+  ``mongosync_reserved_for_internal_use`` system database prior to
+  starting.
+- You cannot modify a filter that is in use. To create a new filter, see
+  :ref:`changing the filter <c2c-change-filter>`.
+- If you rename a collection or database that is part of a filter,
```
Comment (rohshar) [inline source/includes/limitations-filtering.rst:9]: This is confusing. It could continue syncing, if the new namespace is in the filter. There are special cases for within DB and cross DB renames.

File: source/includes/limitations-filtering.rst [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+  ``mongosync`` does not continue syncing under the new name.
+- If you rename a collection or database so that the new name matches
+  part of an existing filter, ``mongosync`` cannot sync the renamed
+  entity.
+- If a filter includes a :ref:`view <views-landing-page>` but not the
+  base collection, only the view is replicated.
+- You cannot specify system collections or system databases.
```
Comment (rohshar) [inline source/includes/limitations-filtering.rst:16]: ..databases in the filter

File: source/includes/limitations-filtering.rst [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+  <c2c-api-reverse>`.
+- The destination cluster must not contain user data prior to starting.
+- The destination cluster must not contain the
+  ``mongosync_reserved_for_internal_use`` system database prior to
+  starting.
+- You cannot modify a filter that is in use. To create a new filter, see
+  :ref:`changing the filter <c2c-change-filter>`.
+- If you rename a collection or database that is part of a filter,
+  ``mongosync`` does not continue syncing under the new name.
+- If you rename a collection or database so that the new name matches
```
Comment (rohshar) [inline source/includes/limitations-filtering.rst:11]: This is incorrect. If a rename happens within the filter within the same DB, then it will continue to sync on the renamed entity.

For renaming to a different DB, there are special rules. See `Renaming Namespaces` in https://docs.google.com/document/d/1vDjeXw8x592d3NLKL5RHL6Nouw64heAL-yod_PyqKME/edit for more info

**Review by rohshar — CHANGES_REQUESTED** [review]
I definitely think we need a section on renaming, where it explains what renames will work, and which ones won't. See https://docs.google.com/document/d/1vDjeXw8x592d3NLKL5RHL6Nouw64heAL-yod_PyqKME/edit under `Renaming Namespaces` for details on renames

File: source/reference/collection-level-filtering.txt [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+      Identify the databases and collections that you want to sync to
+      the destination cluster. When you add a set of databases to the
+      filter, you also exclude any other databases in the cluster.
+      
+      If specify a collection in your filter, you also exclude any other
+      collections that are in the same database.
+
+   .. step:: Create a Filter.
+
+      The ``includeNamespaces`` option specifies a filter that you can
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:49]: ... parameter specifies an optional filter that you can pass to the API. Note that omitting the filter will cause mongosync to do a full cluster sync.

**Review by davemungo — COMMENTED** [review]
@rohshar thanks for the feedback. Would you have another look? 

FYI - Alan also wanted to take a look at the changes so I am going to add him as well.

**Review by alan-zheng-mdb — CHANGES_REQUESTED** [review]

**Review by kennethdyer — APPROVED** [review]
LGTM

File: source/reference/collection-level-filtering.txt [hunk 18]
```diff
@@ -0,0 +1,251 @@
+.. _c2c-collection-filtering:
+
+==========================
+Collection Level Filtering
```
[the remark this answers is not in the document]
Comment (davemungo) [reply source/reference/collection-level-filtering.txt]: Done

File: source/reference/collection-level-filtering.txt [hunk 19]
```diff
@@ -0,0 +1,251 @@
+.. _c2c-collection-filtering:
```
[the remark this answers is not in the document]
Comment (davemungo) [reply source/reference/collection-level-filtering.txt]: Done. 
This is for internal linking, it's not a tab, just a reference

File: source/reference/collection-level-filtering.txt [hunk 20]
```diff
@@ -0,0 +1,251 @@
...
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: twocols
+
+.. versionadded:: 1.1
+
+{+c2c-product-name+} provides continuous data synchronization or a 
+one-time data migration between two MongoDB clusters. You can use
+collection level filtering to control the data that the :ref:`mongosync
```
[the remark this answers is not in the document]
Comment (davemungo) [reply source/reference/collection-level-filtering.txt]: Done

**Review by davemungo — COMMENTED** [review]
@alan-zheng-mdb thanks for your feedback, this is ready for another look

File: source/includes/example-filter-collection.rst [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+         {
+             "database" : "marketing"
+         }
+      ]
+
+After you call the ``/start`` API with this filter in place,
+``mongosync``:
+
+- Syncs all of the collections in the ``marketing`` database
+- Syncs the ``EMEA`` and ``APAC`` collections from the ``sales``
```
Comment (rohshar) [inline source/includes/example-filter-collection.rst:31]: from the sales databases, filtering out the Amer collection

File: source/reference/collection-level-filtering.txt [hunk 22]
```diff
@@ -0,0 +1,251 @@
...
+.. _c2c-example-start-with-filter: 
+
+Start ``mongosync`` with a Filter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/example-filter-collection.rst
+
+.. _c2c-example-rename-while-syncing:
+
+Renaming Collections While Syncing
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:247]: The below section also talks about adding collections

File: source/includes/example-filter-collection-with-renaming.rst [hunk 23]
```diff
@@ -0,0 +1,108 @@
...
+You can rename any collection in the ``staff`` database. 
+
+.. code-block:: shell
+
+   // This code works 
+   use admin
+   db.runCommand( { renameCollection: "staff.employees", to: "staff.salaried" } )
+
+You can only rename a collection within the ``students`` database if the
+new and old names are both in the filter. If both names are not in the
```
Comment (rohshar) [inline source/includes/example-filter-collection-with-renaming.rst:61]: If either name is not

File: source/includes/example-filter-collection-with-renaming.rst [hunk 24]
```diff
@@ -0,0 +1,108 @@
...
+cannot rename it to remove it from the filter.
+
+.. code-block:: shell
+   :copyable: false 
+
+   // This code produces an error and mongosync stops syncing 
+   use admin
+   db.runCommand( { renameCollection: "students.graduate", to: "students.notAFilteredCollection" } )
+
+You can rename a collection to add it to a filter if the whole target
```
Comment (rohshar) [inline source/includes/example-filter-collection-with-renaming.rst:80]: This is confusing as is. Based on what we talked about, I know what you're going for, but it might be better to simply say "you can rename a collection into a different database if the whole target database is included in the filter"

File: source/includes/example-filter-collection-with-renaming.rst [hunk 25]
```diff
@@ -0,0 +1,108 @@
...
+database is included in the filter. 
+
+- Within a database, no collections specified in the filter's target database
```
Comment (rohshar) [inline source/includes/example-filter-collection-with-renaming.rst:83]: Whole target database in the filter, renames within a database

File: source/includes/example-filter-collection-with-renaming.rst [hunk 26]
```diff
@@ -0,0 +1,108 @@
...
+
+  .. code-block:: shell
+
+     use admin
+     db.runCommand( { renameCollection: "staff.employees", to: "staff.onPayroll" } )
+
+- Across databases, no collections specified in the filter's target database
```
Comment (rohshar) [inline source/includes/example-filter-collection-with-renaming.rst:90]: Instead of saying no collections specified, say source collection not specified in the filter, and target database specified in the filter

File: source/includes/example-filter-collection-with-renaming.rst [hunk 27]
```diff
@@ -0,0 +1,108 @@
...
+
+  .. code-block:: shell
+
+     use admin
+     db.runCommand( { renameCollection: "prospects.current", to: "staff.newHires" } )
+
+- Across databases, both collections specified in the filter
```
Comment (rohshar) [inline source/includes/example-filter-collection-with-renaming.rst:97]: Source collection specified, whole target DB specified

**Review by rohshar — COMMENTED** [review]
Definitely looking better!

**Review by davemungo — COMMENTED** [review]
@rohshar thanks for your feedback. Would you have a look at these updates?

File: source/reference/limitations.txt [hunk 28]
```diff
@@ -122,6 +118,13 @@ Multiple Clusters
 - Syncing one source cluster to many destination clusters is not
   supported.
 
+.. _c2c-filtering-limitations:
+
+Collection Level Filtering
```
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:123]: conform to "Filtered Sync"

File: source/reference/api/start.txt [hunk 29]
```diff
@@ -111,16 +111,38 @@ Request Body Parameters
...
+       the filter definition. ``mongosync`` will not sync the other
+       existing databases.
+       
+       If you want to modify the filter to add a newly created database,
+       you have to :ref:`restart the filtered sync <c2c-change-filter>`
+       from the beginning.
+
+       For more details, see: :ref:`c2c-filtered-sync`.
+
+       For current limitations, see: :ref:`c2c-filtering-limitations`.
```
Comment (alan-zheng-mdb) [inline source/reference/api/start.txt:135]: conform to "Filtered Sync"

**Review by alan-zheng-mdb — CHANGES_REQUESTED** [review]
Thanks. Two more places that still mention "Collection Level Filtered". Conform to "Filtered Sync"

File: source/reference/limitations.txt (hunk 28 above)
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/api/start.txt (hunk 29 above)
Comment (davemungo) [reply source/reference/api/start.txt:135]: There's no copy change here. The link pulls in the title from the updated reference below.

**Review by davemungo — COMMENTED** [review]
@alan-zheng-mdb thanks for the feedback. This is ready for another look

File: source/reference/api/start.txt [hunk 30]
```diff
@@ -111,16 +111,38 @@ Request Body Parameters
 
        Default value is ``false``.
 
+   * - ``includeNamespaces``
+     - array
+     - Optional
+     - Filters the databases or collections to sync. The filter document
```
Comment (rohshar) [inline source/reference/api/start.txt:117]: The filter json (document might be confusing)

File: source/reference/collection-level-filtering.txt [hunk 31]
```diff
@@ -0,0 +1,258 @@
...
+      database that you identified in step 1. Use the ``"database"``
+      field to specify the database name.
+      
+      If you want to filter on collections within a database, add those
+      collections to the list in the ``"collections"`` field for that
+      database entry.
+
+   .. step:: Use the Filter.
+
+      To use the filter, attach the filter document when you make the
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:69]: The filter json (document might be confusing)

File: source/reference/collection-level-filtering.txt [hunk 32]
```diff
@@ -0,0 +1,258 @@
...
+
+Replace an Existing Filter
+--------------------------
+
+You cannot update an existing filter. You must stop the ongoing sync
+process, prepare the destination cluster, and restart ``mongosync`` with
+a new filter. 
+
+When ``mongosync`` ran your original filter, it created databases with
+your data, "user databases", and the
```
Comment (rohshar) [inline source/reference/collection-level-filtering.txt:87]: Can you put "user databases" in parenthesis. Otherwise it looks like your data and user dbs are different elements in the comma separated list

File: source/includes/example-filter-collection-with-renaming.rst [hunk 33]
```diff
@@ -0,0 +1,109 @@
...
+target database is in the filter:
+
+.. code-block:: shell
+
+   use admin
+   db.runCommand( { renameCollection: "staff.employees", to: "staff.onPayroll" } )
+
+.. important::
+
+   If you anticipate renaming collections, add the entire database to
```
Comment (rohshar) [inline source/includes/example-filter-collection-with-renaming.rst:107]: "consider adding the"

Not sure we want to make that such a strong suggestion

**Review by rohshar — COMMENTED** [review]
Very close! I'll approve after these changes.

File: source/includes/example-filter-collection-with-renaming.rst (hunk 33 above)
Comment (davemungo) [reply source/includes/example-filter-collection-with-renaming.rst]: Done

File: source/reference/api/start.txt (hunk 30 above)
Comment (davemungo) [reply source/reference/api/start.txt]: Done

File: source/reference/collection-level-filtering.txt (hunk 31 above)
Comment (davemungo) [reply source/reference/collection-level-filtering.txt]: Done

File: source/reference/collection-level-filtering.txt (hunk 32 above)
Comment (davemungo) [reply source/reference/collection-level-filtering.txt]: Done

**Review by davemungo — COMMENTED** [review]
@rohshar Thanks for your feedback, this is ready for another look

**Review by rohshar — APPROVED** [review]

File: source/reference/collection-level-filtering.txt [hunk 34]
```diff
@@ -0,0 +1,258 @@
...
+
+   .. step:: Configure a New Filter.
+
+      a. Decide which databases and collections you want to filter on.
+         Add the databases and collections to the ``includeNamespaces``
+         array. For configuration details, see
+         :ref:`c2c-configure-filter`.
+
+      #. Run ``mongosync`` to :ref:`reconnect <c2c-conn-top-level>` the
+         source and destination clusters.
```
Comment (edobranov) [inline source/reference/collection-level-filtering.txt:148]:
```suggestion
      #. Run ``mongosync`` to :ref:`reconnect <c2c-conn-top-level>` to the
         source and destination clusters.
```

File: source/reference/collection-level-filtering.txt [hunk 35]
```diff
@@ -0,0 +1,258 @@
...
+   ``mongosync`` stops syncing and reports an error.
+
+   To clean up and restart after an error, follow the steps to
+   :ref:`replace an existing filter <c2c-change-filter>`.
+
+Adding and Renaming Within a Single Database
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can add new collections or rename an existing collection if the
+entire database is part of a filter.
```
Comment (edobranov) [inline source/reference/collection-level-filtering.txt:173]:
```suggestion
entire database is part of the filter. 
```

File: source/includes/example-filter-collection-with-renaming.rst [hunk 36]
```diff
@@ -0,0 +1,109 @@
...
+  collections.
+
+The ``includeNamespaces`` array in this example defines a filter on two
+of the databases:
+
+.. code-block:: shell
+
+   {
+      "source": "cluster0",
+      "destination": "cluster1", "includeNamespaces" :
```
Comment (edobranov) [inline source/includes/example-filter-collection-with-renaming.rst:18]:
```suggestion
      "destination": "cluster1",
      "includeNamespaces" :
```
I think it's a bit more conventional to put each JSON field on its own line.

File: source/includes/example-filter-collection-with-renaming.rst [hunk 37]
```diff
@@ -0,0 +1,109 @@
...
+         [
+            { "database" : "students", "collections": ["undergrad", "graduate", "adjuncts"] },
+            { "database" : "staff" }
+         ]
+   }
+
+With this filter in place ``mongosync`` syncs:
```
Comment (edobranov) [inline source/includes/example-filter-collection-with-renaming.rst:25]:
```suggestion
With this filter in place, ``mongosync`` syncs:
```

**Review by edobranov — APPROVED** [review]
Looks like Rohan + others got most the salient points sorted out. This all looks great to me, so I'm just leaving some nits, mostly related to formatting / grammar-isms. Feel free to accept all, some or none of them (and directly resolve any of them). And no need to re-request review after.

Filtered sync definitely has some tedious details involved, so thank you all for your hard work with this! The documentation & examples are read very clearly to me 😄

File: source/reference/collection-level-filtering.txt [hunk 38]
```diff
@@ -0,0 +1,258 @@
...
+         Repeat this step for each user database list:
+
+         .. code-block:: javascript
+          
+            use <user database name>
+            db.dropDatabase()
+
+         Note: After the first :method:`db.dropDatabase()` operation
+         completes, you may need to run it a second time to remove the
+         database.
```
[the remark this answers is not in the document]
Comment (davemungo) [reply source/reference/collection-level-filtering.txt:138]: The admonition is in the docs for dropDatabase. 

Initially I had the command written twice because when I first started experimenting with filtering I had to drop the db twice several times. Later on it seemed to work ok the first. Rohan noted that dropping consistently worked the first time for him. Given that, I decided on a note instead of the doubled command.

File: source/reference/collection-level-filtering.txt [hunk 39]
```diff
@@ -0,0 +1,257 @@
...
+
+   .. step:: Configure a New Filter.
```
[the remark this answers is not in the document]
Comment (davemungo) [reply source/reference/collection-level-filtering.txt]: Done

File: source/reference/collection-level-filtering.txt (hunk 34 above)
Comment (davemungo) [reply source/reference/collection-level-filtering.txt]: Done - I think this works either way, adding 'to' for clarity

File: source/reference/collection-level-filtering.txt (hunk 35 above)
Comment (davemungo) [reply source/reference/collection-level-filtering.txt]: Done

**Review by davemungo — COMMENTED** [review]
@edobranov Thanks for your feedback. I've incorporated the changes. Will push next.
